main.go

```diff
@@ -88,16 +89,46 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
+
+	certsReady := make(chan struct{})
+	if err = cert.CertsManager(mgr, certsReady); err != nil {
+		setupLog.Error(err, "unable to setup cert rotation")
+		os.Exit(1)
+	}
+
+	setupIndexes(mgr)
+
+	go setupControllers(mgr, certsReady)
```

Comment:
why do we have to setup the controllers in a separate go routine?

Reply:
controllers are blocked by certReady chan

Reply:
I get that part. 

I guess the part I missed is that the CertsManager call above registers a reconciler with the manager that gets started only when we invoke `mgr.Start` below

Reply:
+1, can you please add the commend Daniel suggested above?

Reply:
done. please check again

Reply:
thanks!